Libraries/Components/WebView/WebView.ios.js

```diff
@@ -462,6 +462,16 @@ var WebView = React.createClass({
   },
 
   /**
+   * Evaluate JavaScript on the current page.
+   */
+  evaluateJavaScript: async function(script: string): Promise {
+    var escaped = JSON.stringify(script).replace(/\u2028/g, '\\u2028').replace(/\u2029/g, '\\u2029');
+    var wrapped = 'JSON.stringify(eval(' + escaped + '))';
```

Comment:
> Why enforce this `JSON.stringify(eval(` pattern`? Why not just send over the original string?

#1191 has the relevant history (starting with @spicyj's [comment](https://github.com/facebook/react-native/pull/1191#issuecomment-107281435)). For forward compatibility with `WKWebView` (and to be more useful in general), we want the API to return the typed result equivalent to the page-side result.

The naive approach for doing this would be to simply say `var wrapped = 'JSON.stringify(' + script + ')'`, using `JSON.parse()` on the result. That works for most cases, but breaks if the script contains multiple statements (i.e., `'foo'; 'bar';`). `eval()` fixes this by executing all statements in the script *and* returning the result of the last statement, which is exactly what we need to feed the result to `JSON.stringify()`. That means we also need to stringify the script itself so we can use it in `eval`.